add linuxquic implementation and enable it in ci

lxin opened the pull request:
This patchset integrates linuxquic into the quic interop runner and enables it in CI.

- The 1st patch adds linuxquic testing image into the implementation.
- The 2nd patch enables linuxquic in interop CI tests

With this patchset, interop tests CI can now run successfully with additional images, as shown in https://github.com/lxin/quic-interop-runner/actions/runs/22547285220

Thanks to Moritz Buhl for initiating the interop tests for linuxquic.

<!-- Macroscope's pull request summary starts here -->
<!-- Macroscope will only edit the content between these invisible markers, and the markers themselves will not be visible in the GitHub rendered markdown. -->
<!-- If you delete either of the start / end markers from your PR's description, Macroscope will append its summary at the bottom of the description. -->
> [!NOTE]
> ### Add linuxquic implementation and enable it in CI interop tests
> - Adds a `linuxquic` entry to [implementations_quic.json](https://github.com/quic-interop/quic-interop-runner/pull/428/files#diff-0c76a030d8e8cfb37c9b3536695477ec1d6758f30907140fd1c3f6cfa4815138) pointing to `quay.io/lxin/linuxquic-interop:latest`.
> - In [implementations.py](https://github.com/quic-interop/quic-interop-runner/pull/428/files#diff-68b24b5ccc83bee59f0e1318cc340606899826bed2d3e2502416816d957a3b01), `linuxquic` is filtered out of the returned implementations unless `/usr/include/linux/quic.h` is present on the host.
> - Adds a CI step in [interop-test.yml](https://github.com/quic-interop/quic-interop-runner/pull/428/files#diff-f6e386535c20db77aa925c6a8ed443534c486443f73ca333f77ef75493698128) that clones and builds the Linux QUIC kernel module when the client or server is `linuxquic`, ensuring the header file exists before tests run.
>
> <!-- Macroscope's review summary starts here -->
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized b4d500b.</sup>
> <!-- Macroscope's review summary ends here -->
>
<!-- macroscope-ui-refresh -->
<!-- Macroscope's pull request summary ends here -->

lxin committed:
add linuxquic implementation

Add linuxquic-interop docker image in implementations_quic.json, and skip
it if the uapi header file /usr/include/linux/quic.h (kernel module)
doesn't exist in implementations.py.

Signed-off-by: Xin Long <lucien.xin@gmail.com>

larseggert commented:
@lxin I tried to add `linuxquic` to the neqo CI (https://github.com/mozilla/neqo/pull/2507) but all tests are failing. What am I doing wrong?

miyangmia commented:
@larseggert did you install the QUIC kernel module on the host? as it does in the patch: "ci: enable linuxquic in interop tests "

alaendle commented:
@marten-seemann I just wanted to ask if it makes sense to consider adding a new Quic implementation right now, because I've seen that this and other pull requests have been sitting for a while. And please don't misunderstand me, I simply want to avoid potentially creating work for no chance of integration. This shouldn't be interpreted as criticism or an expectation; I'm already happy with this framework and can use it for testing as is. So, thank you very much for providing this.

marten-seemann commented:
@lxin With #466, you should now be able to add your implementation to the JSON file without any need for modifyinhg the workflows.

lxin committed:
ci: enable linuxquic in interop tests

Build and install the Linux QUIC module in CI interop tests when
either the client or server is Linux QUIC, ensuring its availability
in interop CI.

Note if the kernel already supports QUIC (the uapi header file
/usr/include/linux/quic.h exists), it will not clone and build the
Linux QUIC module from lxin/quic repo.

Signed-off-by: Xin Long <lucien.xin@gmail.com>
